lib/services/oidc.go

```diff
@@ -95,6 +53,51 @@ func OIDCClaimsToTraits(claims jose.Claims) map[string][]string {
 	return traits
 }
 
+// GetRedirectURL gets a redirect URL for the given connector. If the connector
+// has a redirect URL which matches the host of the given Proxy address, then
+// that one will be returned. Otherwise, the first URL in the list will be returned.
+func GetRedirectURL(conn types.OIDCConnector, proxyAddr string) (string, error) {
+	if len(conn.GetRedirectURLs()) == 0 {
+		return "", trace.BadParameter("No redirect URLs provided")
+	}
+
+	// If a specific proxyAddr wasn't provided in the oidc
+	// auth request, just use the default redirect URL.
+	if proxyAddr == "" {
```

Comment:
If we have a single redirect URL, I wonder if we should just return it right away also? Probably no need to iterate in this case.